### ISSUE TITLE
[Bug]: Cities Apps: HTTPError: 403 Client Error: Forbidden for url: https://api.citiesapps.com/auth/guest 

### ISSUE THREAD

**Issue by thefox69** [mampfes/hacs_waste_collection_schedule#1424]
### I Have A Problem With:

A specific source

### What's Your Problem

Since one day  an error with cities app happens.
Was running fine for some month.
Maybe Cities app changed again the authentication

### Source (if relevant)

Cities Apps Stockerau

### Logs

```Shell
fetch failed for source App CITIES: Traceback (most recent call last): File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py", 
line 134, in fetch entries = self._source.fetch() ^^^^^^^^^^^^^^^^^^^^ File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source/citiesapps_com.py", 
line 53, in fetch cities_apps = CitiesApps() ^^^^^^^^^^^^ File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/service/CitiesAppsCom.py", 
line 1094, in __init__ r.raise_for_status() File "/usr/local/lib/python3.11/site-packages/requests/models.py", 
line 1021, in raise_for_status raise HTTPError(http_error_msg, response=self) requests.exceptions.HTTPError: 403 Client Error: Forbidden for url: https://api.citiesapps.com/auth/guest
```

### Relevant Configuration

```YAML
#waste_collection_schedule:
#    sources:
#    - name: citiesapps_com
#      args:
#        city: Stockerau
#        calendar: Abfallkalender - Abfuhrzone D
waste_collection_schedule:
    sources:
    - name: citiesapps_com
      args:
        city: Rudersdorf
        calendar: Rudersdorf 3
```

### Checklist Source Error

- [X] Use the example parameters for your source (often available in the documentation) (don't forget to restart Home Assistant after changing the configuration)
- [X] Checked that the website of your service provider is still working
- [X] Tested my attributes on the service provider website (if possible)
- [X] I have tested with the latest version of the integration (master) (for HACS in the 3 dot menu of the integration click on "Redownload" and choose master as version)

### Checklist Sensor Error

- [ ] Checked in the Home Assistant Calendar tab if the event names match the types names (if types argument is used)

### Required

- [X] I have searched past (closed AND opened) issues to see if this bug has already been reported, and it hasn't been.
- [X] I understand that people give their precious time for free, and thus I've done my very best to make this problem as easy as possible to investigate.

**Comment by AlpenFlizzer** [comment]
I have the same issue. This is my error message from the log:

```
2023-11-16 20:11:21.759 ERROR (SyncWorker_3) [waste_collection_schedule.source_shell] fetch failed for source App CITIES:
Traceback (most recent call last):
  File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py", line 134, in fetch
    entries = self._source.fetch()
              ^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source/citiesapps_com.py", line 53, in fetch
    cities_apps = CitiesApps()
                  ^^^^^^^^^^^^
  File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/service/CitiesAppsCom.py", line 1094, in __init__
    r.raise_for_status()
  File "/usr/local/lib/python3.11/site-packages/requests/models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 403 Client Error: Forbidden for url: https://api.citiesapps.com/auth/guest 
```
I issued some requests by hand, obtained an auth token and sent the following POST at https://api.citiesapps.com/auth/guest
```
POST /auth/guest HTTP/1.1
Host: api.citiesapps.com
user-agent: cities/100.100.100/Android
Authorization: Basic NjU1NjZlMWUwY2I1MWEwMDExM2Q2ZWYyOmE4MzEzYmYwYzM2YzhlYjg3NWY1MGYxYmNiZjA0ODM3YTk4NTc1YzhjODMzZWRkMWQ0N2Q5ZDEyY2Y3Mg==
Content-Type: application/json
Content-Length: 0
```
which gave the following response:
```
HTTP/1.1 403 Forbidden
Server: nginx/1.10.3 (Ubuntu)
Date: Thu, 16 Nov 2023 19:44:58 GMT
Content-Type: application/json; charset=utf-8
Transfer-Encoding: chunked
Connection: keep-alive
Vary: Accept-Encoding
X-Powered-By: Express
Access-Control-Allow-Origin: *
Access-Control-Allow-Methods: POST, GET, PUT, DELETE, OPTIONS
Access-Control-Allow-Headers: Origin, X-Requested-With, Content-Type, Accept, Authorization, X-Requesting-App
X-RateLimit-Limit: 3
X-RateLimit-Remaining: 2
X-RateLimit-Reset: 1700165777
ETag: W/"77-Yv1vH6jEwntkYQ0rxN+dX3ZKT50"
Content-Encoding: gzip

{"substatus":null,"message":"Forbidden","minimum_auth_level":"basic","current_auth_level":"basic","additional_info":{}}
```
I wonder if I made something wrong or if the problem has something to do with the JSON contents saying, current auth level is not satisfying the minimum auth level requirements... Even though basic is required and is provided.

Kind of a downer, that citiesapps API is slightly unreliable... Second breaking change on their side within 2 months or so...

**Comment by 5ila5** [comment]
looks like they removed the guest login method and you now need an account. I may be able to redesign some login stuff to login with a provided mail address and password but you probably will need to create an account in the app to use this source. 
I will look into this but it may take some time

**Comment by AlpenFlizzer** [comment]
I agree, an official API would be nice. However I am here for testing, if you happen to implement the login with account. I might also have a deeper look into the authentication part of the API and your code and see if I can provide any usable input...

**Comment by lesgrebe** [comment]
> I agree, an official API would be nice. However I am here for testing, if you happen to implement the login with account. I might also have a deeper look into the authentication part of the API and your code and see if I can provide any usable input...

Hello same here for me in Fürstenfeld (of course, same APP in use).
An login option with account would be great if its possible 👍 

Thx.
Michi


<details><summary>Log-Details</summary>
<p>

Dieser Fehler wurde von einer benutzerdefinierten Integration verursacht

Logger: waste_collection_schedule.source_shell
Source: custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py:136
Integration: waste_collection_schedule (documentation)
First occurred: 11:45:04 (1 occurrences)
Last logged: 11:45:04

fetch failed for source App CITIES: Traceback (most recent call last): File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py", line 134, in fetch entries = self._source.fetch() ^^^^^^^^^^^^^^^^^^^^ File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source/citiesapps_com.py", line 53, in fetch cities_apps = CitiesApps() ^^^^^^^^^^^^ File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/service/CitiesAppsCom.py", line 1094, in __init__ r.raise_for_status() File "/usr/local/lib/python3.11/site-packages/requests/models.py", line 1021, in raise_for_status raise HTTPError(http_error_msg, response=self) requests.exceptions.HTTPError: 403 Client Error: Forbidden for url: https://api.citiesapps.com/auth/guest

</p>
</details>

**Comment by 5ila5** [comment]
Modified the source to provide the ability login in #1448 feel free to take a look at the code.
They did some changes under the hood as well and moved the majourity of requests to an V2 API. The waste collection request is still requested uing the "old v1" API. So this source will break again if they decide to move this to the V2 API as well (I cannot prepare for this as they probably change the structure of the requests/responses)

**Comment by 5ila5** [comment]
fixed by #1448 

fix will be part of release 1.44.0 or is available now as master (git) version (HACS: 3-dot menu -> redownload -> change version to master, don't forget to fully restart HA)
